packages/pds/src/app-view/api/app/bsky/feed/getPostThread.ts

```diff
@@ -344,3 +388,70 @@ const insertIntoThreadReplies = async (
     replies,
   }
 }
+
+const threadPostView = async (
+  localSrvc: LocalService,
+  descript: RecordDescript<PostRecord>,
+): Promise<ThreadViewPost | null> => {
+  const postView = await localSrvc.getPost(descript)
+  if (!postView) return null
+  return {
+    $type: 'app.bsky.feed.defs#threadViewPost',
+    post: postView,
+  }
+}
+
+// Read after write on error
+// ---------------------
+
+const readAfterWriteNotFound = async (
+  ctx: AppContext,
+  params: QueryParams,
+  requester: string,
+  headers?: Headers,
+): Promise<{ data: OutputSchema; lag?: number } | null> => {
+  if (!headers) return null
+  const rev = getRepoRev(headers)
+  if (!rev) return null
+  const uri = new AtUri(params.uri)
+  if (uri.hostname !== requester) {
+    return null
+  }
+  const localSrvc = ctx.services.local(ctx.db)
+  const local = await localSrvc.getRecordsSinceRev(requester, rev)
+  const found = local.posts.find((p) => p.uri.toString() === uri.toString())
+  if (!found) return null
+  let thread = await threadPostView(localSrvc, found)
+  if (!thread) return null
+  const rest = local.posts.filter((p) => p.uri.toString() !== uri.toString())
+  thread = await addPostsToThread(localSrvc, thread, rest)
+  const highestParent = getHighestParent(thread)
+  if (highestParent) {
+    try {
+      const parentsRes = await ctx.appviewAgent.api.app.bsky.feed.getPostThread(
+        { uri: highestParent, parentHeight: params.parentHeight, depth: 0 },
+        await ctx.serviceAuthHeaders(requester),
+      )
+      thread.parent = parentsRes.data.thread
+    } catch (err) {
+      // do nothing
+    }
+  }
+  return {
+    data: {
+      thread,
+    },
+    lag: getLocalLag(local),
+  }
+}
+
+const getHighestParent = (
+  thread: ThreadViewPost,
+  height = 0,
```

Comment:
May be able to get rid of this argument, at least I don't see it getting any use at the moment.